linalg/solvers.hpp

```diff
@@ -115,6 +115,44 @@ class OperatorJacobiSmoother : public Solver
    const Operator *oper;
 };
 
+/// Chebyshev accelerated smoothing with given vector, no matrix necessary
+/** Potentially useful with tensorized operators, for example.
+    This is just a very basic Chebyshev iteration, if you want
+    tolerances, iteration control, etc. wrap this with SLISolver. */
+class OperatorChebyshevSmoother : public Solver
+{
+public:
+   /** Application is by *inverse* of the given vector.
+       It is assumed the underlying operator acts as the identity
+       on entries in ess_tdof_list, corresponding to (assembled) DIAG_ONE
+       policy or ConstrainedOperator in the matrix-free setting. */
+   OperatorChebyshevSmoother(Operator* oper_, const Vector &d,
```

Comment:
You can either run power method or do Gershgorin estimate (should be doable in matrix free (?))

Reply:
I liked the idea of @barker29 and added another constructor which uses the `PowerMethod` internally to estimate the largest eigenvalue. @delyank do you mean to use the Gershgorin circle theorem? If yes, I do not think it is possible to get a tight estimate on the largest eigenvalue with it since you only obtain a set of intervals where the eigenvalues lie in. An alternative to the power method would be the Lanczos algorithm which can be used for symmetric matrices only but has better convergence properties.